jcli/src/jcli_app/vote/mod.rs

```diff
@@ -0,0 +1,103 @@
+mod committee_communication_key;
+mod committee_member_key;
+mod common_reference_string;
+mod encrypting_vote_key;
+
+use structopt::StructOpt;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("I/O error")]
+    Io(#[from] std::io::Error),
+    #[error("invalid Hexadecimal")]
+    Hex(#[from] hex::FromHexError),
+    #[error("error while using random source")]
+    Rand(#[from] rand::Error),
+    #[error("invalid seed length, expected 32 bytes but received {seed_len}")]
+    InvalidSeed { seed_len: usize },
+    #[error("invalid output file path '{path}'")]
+    InvalidOutput {
+        #[source]
+        source: std::io::Error,
+        path: std::path::PathBuf,
+    },
+    #[error("invalid public key")]
+    InvalidPublicKey,
+    #[error("invalid secret key")]
+    InvalidSecretKey,
+    #[error("invalid common reference string")]
+    InvalidCrs,
+    #[error("must provide at least one member key to build encrypting vote key")]
+    EncryptingVoteKeyFromEmpty,
+    #[error("expected at least one committee key")]
+    EmptyCommittee,
+    #[error("threshold should be in range (0..{committee_members:?}] and is {threshold:?}")]
+    InvalidThreshold {
+        threshold: usize,
+        committee_members: usize,
+    },
+    #[error("invalid committee member index")]
+    InvalidCommitteMemberIndex,
+}
+
+#[derive(StructOpt)]
+#[structopt(rename_all = "kebab-case")]
+pub enum Vote {
+    /// Build a commitee communication key
+    CommitteeCommunicationKey(committee_communication_key::CommitteeCommunicationKey),
+    /// Build a committee member key
+    CommitteeMemberKey(committee_member_key::CommitteeMemberKey),
+    /// Build an encryption vote key
+    MakeEncryptingVoteKey(encrypting_vote_key::MakeEncryptingVoteKey),
```

Comment:
I'd like to bikeshed this to a shorter one too. As we are under the `vote` command and these are all "make" commands anyway, this could be `jcli vote encrypting-key ...`